src/pyats/contrib/creators/netbox.py

```diff
@@ -563,19 +603,34 @@ def _generate(self):
             device_id = device["id"]
             device_data = data.setdefault(device_name, {})
 
-            # Construct device platform data
-            device_platform = self._parse_os(self._get_info(device, 
-                            ["platform", "slug"], lambda slug: slug.lower()))
-
-            # OS value is required and must exist
-            is_valid &= self._set_value_if_exists(device_data, "os",
-                                                             device_platform)
+            # If provided, use Platform Map to translate Netbox Platform to PyATS Platform/OS
+            platform_map = self._platform_map
+            platform_map_success = False
+            if platform_map:
+                raw_device_platform = self._get_info(device, 
+                                ["platform", "slug"], lambda slug: slug.lower())
+                if raw_device_platform in platform_map.keys():
+                    # Mapping found in platform_map, set platform and os.
+                    device_platform = platform_map[raw_device_platform]["platform"]
+                    is_valid &= self._set_value_if_exists(device_data, "os",
+                                        platform_map[raw_device_platform]["os"])
+                    platform_map_success = True
+
+            if platform_map_success == False:
```

Comment:
I think a small comment here to mention that this is default behavior would be useful for future reference. Something kinda like (please adjust to your liking):
```suggestion
            # If platform mapping was not provided or was unsuccessful, then default to standard behavior for assigning os and platform
            if platform_map_success == False:
```